torch/testing/_internal/common_methods_invocations.py

```diff
@@ -6256,6 +6299,77 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            supports_forward_ad=True,
            supports_out=False,
            autodiff_nonfusible_nodes=["aten::hardswish"]),
+    OpInfo('nn.functional.interpolate',
+           aten_name="interpolate",
+           variant_test_name='nearest',
```

Comment:
Is the purpose of adding multiple interpolation-type variants because:
* Different dtypes are supported for each type of interpolation
* Some interpolation types are supported on CUDA and some aren't
* Any other reason?

Reply:
The main reason is that each variant calls a different backend: https://github.com/pytorch/pytorch/blob/a8f6b5a80a48ed814443c36d40f9eaf5c6aef7f6/torch/nn/functional.py#L3689-L3708 .

This leads to weird behavior that different dtypes are supported for each type of interpolation on different devices

Reply:
Gotcha- thanks for the clarification. This is a pretty interesting case for ModuleInfo & functional OpInfo generation - the single module `nn.Upsample` encapsulates all these different ops with different dtype / device support.